### PR TITLE
Add world access and change endpointSelector

### PR DESCRIPTION
**Pull request by fhielpos**
Closes: https://github.com/giantswarm/giantswarm/issues/31210

### CODE REVIEW

File: helm/trivy-operator/templates/ciliumnetworkpolicy.yaml [hunk 1]
```diff
@@ -39,14 +39,18 @@ metadata:
...
   endpointSelector:
-    matchLabels:
-      {{- include "scannerjob.labels.selector" . | nindent 6 }}
+    matchExpressions:
+    - key: vulnerabilityReport.scanner
+      operator: In
+      values:
+      - Trivy
   egress:
-  {{ if eq (index .Values "trivy-operator" "trivy" "mode") "Standalone"}}
```
Comment (stone-z) [inline helm/trivy-operator/templates/ciliumnetworkpolicy.yaml:45]: How hard would it be to keep it gated to only standalone or the scan modes that actually need policies?

Comment (fhielpos) [reply helm/trivy-operator/templates/ciliumnetworkpolicy.yaml:45]: Apparently not, couldn't find anything gating this

File: helm/trivy-operator/templates/ciliumnetworkpolicy.yaml [hunk 2]
```diff
@@ -39,14 +39,18 @@ metadata:
...
-    # Allow egress to the internet to fetch database updates.
-    - toEntities:
-        - world
-  {{- end -}}
+    # Allow egress to the internet to fetch database updates and trivy-checks OCI.
+    - toFQDNs:
+        - matchName: "gsoci.azurecr.io"
```
Comment (stone-z) [inline helm/trivy-operator/templates/ciliumnetworkpolicy.yaml:50]: Sorry for nitpicking -- does this need to be localized for China as well?

**Comment by stone-z** [comment]
Also, the regular NetworkPolicy should follow and we can ditch the helper https://github.com/search?q=repo%3Agiantswarm%2Ftrivy-operator-app%20scannerJob.labels&type=code

File: helm/trivy-operator/templates/ciliumnetworkpolicy.yaml (hunk 2 above)
Comment (fhielpos) [reply helm/trivy-operator/templates/ciliumnetworkpolicy.yaml:50]: I'm trying to make it work first, it seems that just allowing this still doesn't grant the access for DNS

**Review by ssyno — APPROVED** [review]

**Comment by fhielpos** [comment]
Resulting CNP with default values:
```yaml
spec:
  endpointSelector:
    matchLabels:
      vulnerabilityReport.scanner: Trivy
  egress:
    # Allow egress to the internet to fetch database updates and trivy-checks OCI.
    - toPorts:
        - ports:
          - port: "443"
            protocol: TCP
      toEntities:
        - world
    # Allow port 80 connection to instance metadata
    - toCIDR:
      - "169.254.169.254/32"
      toPorts:
      - ports:
        - port: "80"
          protocol: TCP
    # Allow Trivy service connection
    - toEntities:
        - cluster
      toPorts:
        - ports:
          - port: "4954"
    # Allow DNS for FQDN resolution
    - toEndpoints:
      - matchLabels:
          "k8s:io.kubernetes.pod.namespace": kube-system
          "k8s:k8s-app": coredns
      - matchLabels:
          "k8s:io.kubernetes.pod.namespace": kube-system
          "k8s:k8s-app": k8s-dns-node-cache
      toPorts:
      - ports:
        - port: "1053"
          protocol: ANY
        - port: "53"
          protocol: ANY
```

Resulting CNP with extra FQDNs values:
```yaml
spec:
  endpointSelector:
    matchLabels:
      vulnerabilityReport.scanner: Trivy
  egress:
    # Allow egress to the internet to fetch database updates and trivy-checks OCI.
    - toPorts:
        - ports:
          - port: "443"
            protocol: TCP
      toFQDNs:
      - matchName: gsoci.azurecr.io
      - matchPattern: '*.blob.core.windows.net'
      - matchName: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
      - matchPattern: '*.aliyuncs.com'
      # Allow additional FQDNs rules
      - matchPattern: '*.my-custom-cr.com'
    # Allow port 80 connection to instance metadata
    - toCIDR:
      - "169.254.169.254/32"
      toPorts:
      - ports:
        - port: "80"
          protocol: TCP
    # Allow Trivy service connection
    - toEntities:
        - cluster
      toPorts:
        - ports:
          - port: "4954"
    # Allow DNS for FQDN resolution
    - toEndpoints:
      - matchLabels:
          "k8s:io.kubernetes.pod.namespace": kube-system
          "k8s:k8s-app": coredns
      - matchLabels:
          "k8s:io.kubernetes.pod.namespace": kube-system
          "k8s:k8s-app": k8s-dns-node-cache
      toPorts:
      - ports:
        - port: "1053"
          protocol: ANY
        - port: "53"
          protocol: ANY
        rules:
          dns:
          - matchName: gsoci.azurecr.io
          - matchPattern: '*.blob.core.windows.net'
          - matchName: giantswarm-registry.cn-shanghai.cr.aliyuncs.com
          - matchPattern: '*.cn-shanghai.cr.aliyuncs.com'
          - matchPattern: '*.cn-shanghai.aliyuncs.com'
          - matchPattern: '*.oss-cn-shanghai.aliyuncs.com'
          # Allow additional DNS rules
          - matchPattern: '*.my-custom-cr.com'
```

File: helm/trivy-operator/templates/networkpolicy.yaml [hunk 3]
```diff
@@ -46,15 +46,37 @@ spec:
...
+            kubernetes.io/metadata.name: {{ .Release.Namespace }}
     # Allow egress to the internet to fetch database updates.
-    - to:
+    - ports:
+        - port: 443
+          protocol: TCP
+      to:
       - ipBlock:
           cidr: 0.0.0.0/0
+          except:
```
Comment (stone-z) [inline helm/trivy-operator/templates/networkpolicy.yaml:64]: nice

**Review by stone-z — APPROVED** [review]
Bam

**Pull request merged**